--- iptables/iptables/Makefile.am
+++ iptables/iptables/Makefile.am
@@ -43,8 +43,17 @@ iptables-extensions.8: iptables-extensions.8.tmpl ../extensions/matches.man ../e
 		-e '/@MATCH@/ r ../extensions/matches.man' \
 		-e '/@TARGET@/ r ../extensions/targets.man' $< >$@;
 
+JHOME=/usr/lib/jvm/java-1.6.0-openjdk-1.6.0.41.x86_64
+
 pkgconfig_DATA = xtables.pc
 
+lib_LTLIBRARIES = libtransite.la
+
+libtransite_la_LIBADD = ../libiptc/libip4tc.la ../libiptc/libip6tc.la ../libxtables/libxtables.la
+libtransite_la_CFLAGS = -Wall -fPIC -D_JNI_IMPLEMENTATION_ -I${JHOME}/include -I${JHOME}/include/linux
+libtransite_la_SOURCES = ${xtables_multi_SOURCES} com_ymbl_smartgateway_transite_TransiteActivator.c
+libtransite_la_LDFLAGS = -version-info 1:0:1
+
 # Using if..fi avoids an ugly "error (ignored)" message :)
 install-exec-hook:
 	-if test -z "${DESTDIR}"; then /sbin/ldconfig; fi;
--- iptables/iptables/iptables-standalone.c
+++ iptables/iptables/iptables-standalone.c
@@ -58,7 +58,7 @@
 	init_extensions4();
 #endif
 
-	ret = do_command4(argc, argv, &table, &handle, false);
+	ret = do_command4(argc, argv, &table, &handle, true);
 	if (ret) {
 		ret = iptc_commit(handle);
 		iptc_free(handle);
@@ -78,5 +78,5 @@
 		}
 	}
 
-	exit(!ret);
+	return(!ret);
 }
