--- ppp/chat/Makefile.linux
+++ ppp/chat/Makefile.linux
@@ -25,7 +25,7 @@ chat.o:	chat.c
 
 install: chat
 	mkdir -p $(BINDIR) $(MANDIR)
-	$(INSTALL) -s -c chat $(BINDIR)
+	$(INSTALL) -c chat $(BINDIR)
 	$(INSTALL) -c -m 644 chat.8 $(MANDIR)
 
 clean:
--- ppp/pppd/Makefile.linux
+++ ppp/pppd/Makefile.linux
@@ -48,7 +48,7 @@ MPPE=y
 # Uncomment the next line to include support for PPP packet filtering.
 # This requires that the libpcap library and headers be installed
 # and that the kernel driver support PPP packet filtering.
-FILTER=y
+#FILTER=y
 
 # Uncomment the next line to enable multilink PPP (enabled by default)
 # Linux distributions: Please leave multilink ENABLED in your builds
@@ -102,7 +102,7 @@ ifdef USE_SRP
 CFLAGS	+= -DUSE_SRP -DOPENSSL -I/usr/local/ssl/include
 LIBS	+= -lsrp -L/usr/local/ssl/lib -lcrypto
 TARGETS	+= srp-entry
-EXTRAINSTALL = $(INSTALL) -s -c -m 555 srp-entry $(BINDIR)/srp-entry
+EXTRAINSTALL = $(INSTALL) -c -m 555 srp-entry $(BINDIR)/srp-entry
 MANPAGES += srp-entry.8
 EXTRACLEAN += srp-entry.o
 NEEDDES=y
@@ -208,7 +208,7 @@ all: $(TARGETS)
 install: pppd
 	mkdir -p $(BINDIR) $(MANDIR)
 	$(EXTRAINSTALL)
-	$(INSTALL) -s -c -m 555 pppd $(BINDIR)/pppd
+	$(INSTALL) -c -m 555 pppd $(BINDIR)/pppd
 	if chgrp pppusers $(BINDIR)/pppd 2>/dev/null; then \
 	  chmod o-rx,u+s $(BINDIR)/pppd; fi
 	$(INSTALL) -c -m 444 pppd.8 $(MANDIR)
--- ppp/pppd/plugins/radius/Makefile.linux
+++ ppp/pppd/plugins/radius/Makefile.linux
@@ -36,9 +36,9 @@ all: $(PLUGIN)
 
 install: all
 	$(INSTALL) -d -m 755 $(LIBDIR)
-	$(INSTALL) -s -c -m 755 radius.so $(LIBDIR)
-	$(INSTALL) -s -c -m 755 radattr.so $(LIBDIR)
-	$(INSTALL) -s -c -m 755 radrealms.so $(LIBDIR)
+	$(INSTALL) -c -m 755 radius.so $(LIBDIR)
+	$(INSTALL) -c -m 755 radattr.so $(LIBDIR)
+	$(INSTALL) -c -m 755 radrealms.so $(LIBDIR)
 	$(INSTALL) -c -m 444 pppd-radius.8 $(MANDIR)
 	$(INSTALL) -c -m 444 pppd-radattr.8 $(MANDIR)
 
--- ppp/pppd/plugins/rp-pppoe/Makefile.linux
+++ ppp/pppd/plugins/rp-pppoe/Makefile.linux
@@ -43,7 +43,7 @@ rp-pppoe.so: plugin.o discovery.o if.o common.o
 
 install: all
 	$(INSTALL) -d -m 755 $(LIBDIR)
-	$(INSTALL) -s -c -m 4550 rp-pppoe.so $(LIBDIR)
+	$(INSTALL) -c -m 4550 rp-pppoe.so $(LIBDIR)
 	$(INSTALL) -d -m 755 $(BINDIR)
 	$(INSTALL) -s -c -m 555 pppoe-discovery $(BINDIR)
 
--- ppp/pppdump/Makefile.linux
+++ ppp/pppdump/Makefile.linux
@@ -17,5 +17,5 @@ clean:
 
 install:
 	mkdir -p $(BINDIR) $(MANDIR)
-	$(INSTALL) -s -c pppdump $(BINDIR)
+	$(INSTALL) -c pppdump $(BINDIR)
 	$(INSTALL) -c -m 444 pppdump.8 $(MANDIR)
--- ppp/pppstats/Makefile.linux
+++ ppp/pppstats/Makefile.linux
@@ -22,7 +22,7 @@ all: pppstats
 
 install: pppstats
 	-mkdir -p $(MANDIR)
-	$(INSTALL) -s -c pppstats $(BINDIR)
+	$(INSTALL) -c pppstats $(BINDIR)
 	$(INSTALL) -c -m 444 pppstats.8 $(MANDIR)
 
 pppstats: $(PPPSTATSRCS)
--- ppp/pppd/pathnames.h
+++ ppp/pppd/pathnames.h
@@ -14,6 +14,7 @@
 #define _PATH_DEVNULL	"/dev/null"
 #endif /* HAVE_PATHS_H */
 
+#define _ROOT_PATH "/tmp/transite-target"
 #ifndef _ROOT_PATH
 #define _ROOT_PATH
 #endif
--- ppp/pppd/chap_ms.c
+++ ppp/pppd/chap_ms.c
@@ -81,7 +81,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <ctype.h>
+//#include <ctype.h>
 #include <sys/types.h>
 #include <sys/time.h>
 #include <unistd.h>
--- ppp/pppd/ipxcp.c
+++ ppp/pppd/ipxcp.c
@@ -51,7 +51,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
-#include <ctype.h>
+//#include <ctype.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
--- ppp/pppd/main.c
+++ ppp/pppd/main.c
@@ -69,7 +69,7 @@
 #define RCSID	"$Id: main.c,v 1.156 2008/06/23 11:47:18 paulus Exp $"
 
 #include <stdio.h>
-#include <ctype.h>
+//#include <ctype.h>
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
--- ppp/pppd/multilink.c
+++ ppp/pppd/multilink.c
@@ -28,7 +28,7 @@
  * OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 #include <string.h>
-#include <ctype.h>
+//#include <ctype.h>
 #include <stdlib.h>
 #include <netdb.h>
 #include <errno.h>
--- ppp/pppd/options.c
+++ ppp/pppd/options.c
@@ -42,7 +42,7 @@
 
 #define RCSID	"$Id: options.c,v 1.102 2008/06/15 06:53:06 paulus Exp $"
 
-#include <ctype.h>
+//#include <ctype.h>
 #include <stdio.h>
 #include <errno.h>
 #include <unistd.h>
--- ppp/pppd/plugins/pppoatm/text2atm.c
+++ ppp/pppd/plugins/pppoatm/text2atm.c
@@ -8,7 +8,7 @@
 #endif
 
 #include <stdio.h>
-#include <ctype.h>
+//#include <ctype.h>
 #include <string.h>
 #include <limits.h>
 
--- ppp/pppd/plugins/pppoatm/text2qos.c
+++ ppp/pppd/plugins/pppoatm/text2qos.c
@@ -10,7 +10,7 @@
 #include <stdlib.h>
 #include <stdarg.h>
 #include <string.h>
-#include <ctype.h>
+//#include <ctype.h>
 #include <limits.h>
 
 #include "atm.h"
--- ppp/pppd/plugins/radius/includes.h
+++ ppp/pppd/plugins/radius/includes.h
@@ -16,7 +16,7 @@
 
 #include <sys/types.h>
 
-#include <ctype.h>
+//#include <ctype.h>
 #include <stdio.h>
 #include <errno.h>
 #include <netdb.h>
--- ppp/pppd/plugins/rp-pppoe/debug.c
+++ ppp/pppd/plugins/rp-pppoe/debug.c
@@ -20,7 +20,7 @@ static char const RCSID[] =
 #include <sys/time.h>
 #include <time.h>
 #include <unistd.h>
-#include <ctype.h>
+//#include <ctype.h>
 
 /**********************************************************************
 *%FUNCTION: dumpHex
--- ppp/pppd/plugins/winbind.c
+++ ppp/pppd/plugins/winbind.c
@@ -52,7 +52,7 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <errno.h>
-#include <ctype.h>
+//#include <ctype.h>
 
 #define BUF_LEN 1024
 
--- ppp/pppd/srp-entry.c
+++ ppp/pppd/srp-entry.c
@@ -33,7 +33,7 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <string.h>
-#include <ctype.h>
+//#include <ctype.h>
 #include <t_pwd.h>
 
 #ifndef	SOL2
--- ppp/pppd/sys-linux.c
+++ ppp/pppd/sys-linux.c
@@ -89,7 +89,7 @@
 #include <mntent.h>
 #include <signal.h>
 #include <fcntl.h>
-#include <ctype.h>
+//#include <ctype.h>
 #include <termios.h>
 #include <unistd.h>
 
--- ppp/pppd/sys-solaris.c
+++ ppp/pppd/sys-solaris.c
@@ -92,7 +92,7 @@
 #include <stddef.h>
 #include <stdlib.h>
 #include <string.h>
-#include <ctype.h>
+//#include <ctype.h>
 #include <errno.h>
 #include <fcntl.h>
 #include <unistd.h>
--- ppp/pppd/tty.c
+++ ppp/pppd/tty.c
@@ -71,7 +71,7 @@
 #define RCSID	"$Id: tty.c,v 1.27 2008/07/01 12:27:56 paulus Exp $"
 
 #include <stdio.h>
-#include <ctype.h>
+//#include <ctype.h>
 #include <stdlib.h>
 #include <string.h>
 #include <termios.h>
--- ppp/pppd/utils.c
+++ ppp/pppd/utils.c
@@ -31,7 +31,7 @@
 #define RCSID	"$Id: utils.c,v 1.25 2008/06/03 12:06:37 paulus Exp $"
 
 #include <stdio.h>
-#include <ctype.h>
+//#include <ctype.h>
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
