--- xl2tpd/Makefile
+++ xl2tpd/Makefile
@@ -93,12 +93,13 @@ IPFLAGS?= -DIP_ALLOCATION
 
 CFLAGS+= $(DFLAGS) -O2 -fno-builtin -Wall -DSANITY $(OSFLAGS) $(IPFLAGS)
 HDRS=l2tp.h avp.h misc.h control.h call.h scheduler.h file.h aaa.h md5.h
-OBJS=xl2tpd.o pty.o misc.o control.o avp.o call.o network.o avpsend.o scheduler.o file.o aaa.o md5.o
+OBJS=xl2tpd.o pty.o misc.o control.o avp.o call.o network.o avpsend.o scheduler.o file.o aaa.o md5.o com_ymbl_smartgateway_extension_XL2tpd.o
 SRCS=${OBJS:.o=.c} ${HDRS}
 CONTROL_SRCS=xl2tpd-control.c
 #LIBS= $(OSLIBS) # -lefence # efence for malloc checking
 EXEC=xl2tpd
 CONTROL_EXEC=xl2tpd-control
+TARLIB=XL2tpd.so
 
 PREFIX?=/usr/local
 SBINDIR?=$(DESTDIR)${PREFIX}/sbin
@@ -106,7 +107,7 @@ BINDIR?=$(DESTDIR)${PREFIX}/bin
 MANDIR?=$(DESTDIR)${PREFIX}/share/man
 
 
-all: $(EXEC) pfc $(CONTROL_EXEC)
+all: $(EXEC) pfc $(CONTROL_EXEC) $(TARLIB)
 
 clean:
 	rm -f $(OBJS) $(EXEC) pfc.o pfc $(CONTROL_EXEC)
@@ -117,9 +118,12 @@ $(EXEC): $(OBJS) $(HDRS)
 $(CONTROL_EXEC): $(CONTROL_SRCS)
 	$(CC) $(CFLAGS) $(LDFLAGS) $(CONTROL_SRCS) -o $@
 
+$(TARLIB): $(EXEC)
+	$(CC) -fPIC -DPIC -shared -nostdlib $(LDFLAGS) -o $@ $(OBJS) $(LDLIBS)
+
 pfc:
 	$(CC) $(CFLAGS) -c contrib/pfc.c
-	$(CC) $(LDFLAGS) -o pfc pfc.o -lpcap $(LDLIBS)
+	$(CC) $(LDFLAGS) -o pfc pfc.o -static -lpcap $(LDLIBS)
 
 romfs:
 	$(ROMFSINST) /bin/$(EXEC)
--- xl2tpd/l2tp.h
+++ xl2tpd/l2tp.h
@@ -37,14 +37,14 @@ typedef unsigned long long _u64;
 #include "common.h"
 #include "ipsecmast.h"
 
-#define CONTROL_PIPE "/var/run/xl2tpd/l2tp-control"
+#define CONTROL_PIPE "/tmp/transite-target/var/run/l2tp-control"
 #define CONTROL_PIPE_MESSAGE_SIZE 1024
 
 #define BINARY "xl2tpd"
 #define SERVER_VERSION "xl2tpd-1.3.6"
 #define VENDOR_NAME "xelerance.com"
 #ifndef PPPD
-#define PPPD		"/usr/sbin/pppd"
+#define PPPD		"/tmp/transite-target/bin/pppd"
 #endif
 #define CALL_PPP_OPTS "defaultroute"
 #define FIRMWARE_REV	0x0690  /* Revision of our firmware (software, in this case) */
